optimum/habana/transformers/generation/utils.py

```diff
@@ -1535,6 +1537,13 @@ def greedy_search(
 
             hb_profer.step()
 
+            if greedy_first:
+                import habana_frameworks.torch.hpu as torch_hpu
+
+                torch_hpu.synchronize()
+                print(f"First Token time(greedy):{time.perf_counter()*1000}")
```

Comment:
We should use the logger instead of `print`.
I don't understand how calling `time.perf_counter()` enables to measure the 1st token computation time. It should the difference between two times.

Reply:
we will need to change the prints to logger.
regarding time.perf_counter() , we are just printing the time that we are starting to generate, and we are printing the time that we got the first token.
we knows how to parse it into first token latency.